tests/testthat/test-renderer4-prostateLasso.R

```diff
@@ -0,0 +1,112 @@
+addY <- function(dt, y){
+  data.frame(dt, y.var=factor(y, c("error", "weights")))
+}
+viz.no.time <- list(
+  title="both .variable .value aesthetics",
+  path=ggplot()+
+    theme_bw()+
+    theme(panel.margin=grid::unit(0, "lines"))+
+    facet_grid(y.var ~ ., scales="free")+
+    ylab("")+
+    scale_color_manual(values=variable.colors)+
+    geom_line(aes(arclength, standardized.coef, color=variable, group=variable),
+              data=addY(prostateLasso$path, "weights"))+
+    geom_line(aes(arclength, mse, linetype=set, group=set),
+              data=addY(prostateLasso$error, "error"))+
+    geom_tallrect(aes(
+      xmin=arclength.click-rect.width,
+      xmax=arclength.click+rect.width,
+      id=paste0("arclength", round(arclength.click, 1)*10),
+      clickSelects.variable="arclength",
+      clickSelects.value=arclength.click,
+      showSelected.variable="arclength",
+      showSelected.value=arclength.show,
+      key=ifelse(
+        arclength.click==arclength.show, 1,
+        paste(arclength.click, arclength.show))),
+      alpha=0.5,
+      data=addY(tallrect.all, "error")),
+  res=ggplot()+
+    geom_hline(aes(yintercept=residual),
+               data=hline.df,
+               color="grey")+
+    guides(linetype="none")+
+    geom_point(aes(response, residual, 
+                   key=observation.i,
+                   showSelected=arclength),
+               shape=21,
+               fill=NA,
+               color="black",
+               data=prostateLasso$residuals)+
+    geom_segment(aes(response, residual,
+                     xend=response, yend=0,
+                     linetype=set,
+                     key=observation.i,
+                     showSelected=arclength),
+                 data=prostateLasso$residuals),
+  first=list(arclength=max(arclength)),
+  duration=list(arclength=5000))
+info <- animint2HTML(viz.no.time)
+
+clickID("arclength0")
+Sys.sleep(1)
+html.during <- getHTML()
+Sys.sleep(5)
+html.after <- getHTML()
+
+getGreyRect <- function(html){
+  xpath <- '//g[@class="geom3_tallrect_path"]//rect'
+  node.list <- getNodeSet(html, xpath)
+  opacity.str <- getStyleValue(html, xpath, "opacity")
+  opacity.num <- as.numeric(opacity.str)
+  grey.i <- which(opacity.num == 0.5)
+  node.list[[grey.i]]
+}
+getGreyX <- function(html){
+  grey.rect <- getGreyRect(html)
+  attr.vec <- xmlAttrs(grey.rect)
+  as.numeric(attr.vec[["x"]])
+}
+
+test_that("selected tallrect moves to the left", {
+  x.before <- getGreyX(info$html)
+  x.during <- getGreyX(html.during)
+  expect_lt(x.during, x.before)
+  x.after <- getGreyX(html.after)
+  expect_lt(x.after, x.during)
+})
+
+clickID("arclength174")
+html.click2 <- getHTML()
+
+test_that("tallrect displays correct tooltip", {
+  r <- getGreyRect(html.click2)
+  child.list <- xmlChildren(r)
+  expect_identical(names(child.list), "title")
+  value.vec <- sapply(child.list, xmlValue)
+  expect_identical(paste(value.vec), "arclength 17.4461019561232")
```

Comment:
this tooltip was incorrectly displaying "arclength 0" since the `<title>` element was not being deleted (the old code was adding a second title element with the correct tooltip value, but the browser only displays a tooltip for the first title element)